tests/test_web/test_broker_error.py

```diff
@@ -0,0 +1,218 @@
+
+    assert rv.status_code == 500, rv.json
+    assert mock_hrr.apply_async.call_count == 0
+    assert mock_har.apply_async.call_count == 1
+
+    assert db.session.query(RequestAdd).get(1)
+    assert db.session.query(RequestRm).get(2)
+
+    response_json = {'error': 'The scheduling of the build requests with IDs 1, 2 failed'}
+    assert rv.json == response_json
+
+    requests_to_send_msgs_for = mock_smfnbor.call_args[0][0]
+    assert len(requests_to_send_msgs_for) == 2
+    assert requests_to_send_msgs_for[0].id == 1
+    assert requests_to_send_msgs_for[1].id == 2
+
+    req_states = (
+        db.session.query(RequestState)
+        .join(Request, Request.request_state_id == RequestState.id)
+        .filter(Request.id.in_((1, 2, 3)))
+        .all()
+    )
+    for r in req_states:
+        assert r.state == RequestStateMapping.failed.value
+
+
+@mock.patch('iib.web.api_v1.handle_add_request')
+@mock.patch('iib.web.api_v1.handle_rm_request')
+@mock.patch('iib.web.api_v1.messaging.send_messages_for_new_batch_of_requests')
+def test_add_rm_batch_rm_failure(mock_smfnbor, mock_hrr, mock_har, app, auth_env, client, db):
+    mock_hrr.apply_async.side_effect = OperationalError
+
+    annotations = {'msdhoni': 'Who is that guy?'}
+    data = {
+        'annotations': annotations,
+        'build_requests': [
+            {
+                'bundles': ['registry-proxy/rh-osbs/lgallett-bundle:v1.0-9'],
+                'binary_image': 'registry-proxy/rh-osbs/openshift-ose-operator-registry:v4.5',
+                'from_index': 'registry-proxy/rh-osbs-stage/iib:v4.5',
+                'add_arches': ['amd64'],
+                'cnr_token': 'no_tom_brady_anymore',
+                'organization': 'hello-operator',
+                'overwrite_from_index': True,
+                'overwrite_from_index_token': 'some_token',
+            },
+            {
+                'operators': ['kiali-ossm'],
+                'binary_image': 'registry-proxy/rh-osbs/openshift-ose-operator-registry:v4.5',
+                'from_index': 'registry:8443/iib-build:11',
+            },
+        ],
+    }
+    rv = client.post('/api/v1/builds/add-rm-batch', json=data, environ_base=auth_env)
+
+    assert rv.status_code == 500, rv.json
+    assert mock_hrr.apply_async.call_count == 1
+    assert mock_har.apply_async.call_count == 1
+
+    response_json = {'error': 'The scheduling of the build requests with IDs 2 failed'}
+    assert rv.json == response_json
+
+    requests_to_send_msgs_for = mock_smfnbor.call_args[0][0]
+    assert len(requests_to_send_msgs_for) == 2
+    assert requests_to_send_msgs_for[0].id == 1
+    assert requests_to_send_msgs_for[1].id == 2
+
+    req_add = db.session.query(RequestAdd).get(1)
+    req_rm = db.session.query(RequestRm).get(2)
+    # First request is processed because we are testing failing on RequestRM
+    assert req_add.state.state == RequestStateMapping.in_progress.value
```

Comment:
This approach is cleaner than doing an additional query in `test_add_rm_batch_add_failure`. Could you please adjust it in `test_add_rm_batch_add_failure`?